routing/endpointregistry_test.go

```diff
@@ -200,3 +309,44 @@ func BenchmarkGetInflightRequests(b *testing.B) {
 		benchmarkGetInflightRequests(b, fmt.Sprintf("%d goroutines", goroutines), goroutines)
 	}
 }
+
+func benchmarkGetDetectedTime(b *testing.B, name string, goroutines int) {
+	const key string = "some key"
+	const mapSize int = 10000
+
+	b.Run(name, func(b *testing.B) {
+		r := routing.NewEndpointRegistry(routing.RegistryOptions{})
+		now := time.Now()
+		for i := 1; i < mapSize; i++ {
+			r.GetMetrics(fmt.Sprintf("foo-%d", i)).IncInflightRequest()
+		}
+		r.GetMetrics(key).IncInflightRequest()
+		r.GetMetrics(key).SetDetected(now)
+
+		var dummy time.Time
+		wg := sync.WaitGroup{}
+		b.ResetTimer()
+		b.ReportAllocs()
+		for i := 0; i < goroutines; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				metrics := r.GetMetrics(key)
+				for n := 0; n < b.N/goroutines; n++ {
+					dummy = metrics.DetectedTime()
+				}
+			}()
+		}
+		dummy = dummy.Add(time.Second)
+		wg.Wait()
+
+		printTotalMutexWaitTime(b)
+	})
+}
+
+func BenchmarkGetDetectedTime(b *testing.B) {
+	goroutinesNums := []int{1, 2, 3, 4, 5, 6, 7, 8, 12, 16, 24, 32, 48, 64, 128, 256, 512, 768, 1024, 1536, 2048, 4096}
```

Comment:
Why stop at 4096 and not 8192?

I see one cluster has regular close to 10k goroutines and one outlier cluster that had in the last 8h 31k.
So maybe 32768 makes even sense to test.

Reply:
The benchmarks are fine up to 32k goroutines, I will update results and this code in the next PR.